plugins/de.cau.cs.kieler.klighd.lsp/src/de/cau/cs/kieler/klighd/lsp/interactive/rectpack/RectPackInterativeLanguageServerExtension.xtend

```diff
@@ -0,0 +1,179 @@
+        val parent = kNode.parent
+        val List<KNode> changedNodes = newArrayList(kNode)
+        if (parent.getProperty(CoreOptions.INTERACTIVE_LAYOUT)) {
+            // Sort all nodes regarding their actual position
+            val KNode[] clonedChildren = parent.children.clone()
+            Arrays.sort(clonedChildren, [a,b |
+                return Integer.compare(a.getProperty(RectPackingOptions.CURRENT_POSITION), b.getProperty(RectPackingOptions.CURRENT_POSITION))
+            ])
+
+            // Get previous and new position of node
+            val desiredPosition = parent.children.indexOf(kNode)
+            val int previousPosition = kNode.getProperty(RectPackingOptions.CURRENT_POSITION)
+
+            // Change desired position of nodes that are influenced by the movement
+            val modifier = previousPosition < desiredPosition ? 1 : -1
+            for (var int position = previousPosition; position != desiredPosition; position += modifier) {
+                val node = clonedChildren.get(position)
+                if (node.hasProperty(RectPackingOptions.DESIRED_POSITION)) {
+                    val nodePosition = node.getProperty(RectPackingOptions.DESIRED_POSITION)
+                    if ((previousPosition < nodePosition && nodePosition <= desiredPosition) || 
+                        (previousPosition > nodePosition && nodePosition >= desiredPosition)) {
+                        node.setProperty(
+                            RectPackingOptions.DESIRED_POSITION,
+                            position - modifier)
+                        changedNodes.add(node)
+                    }
+                }
+            }
+            kNode.setProperty(RectPackingOptions.DESIRED_POSITION, null)
+            refreshModelInEditor(changedNodes, uri)
+
+        }
+    }
+    
+    /**
+     * Sets the aspect ratio.
+     */
+    def setAspectRatio(SetAspectRatio constraint, String clientId) {
+        val uri = diagramState.getURIString(clientId)
+        val kNode = LSPUtil.getKNode(diagramState, uri, constraint.id)
+        kNode.setProperty(RectPackingOptions.ASPECT_RATIO, constraint.aspectRatio)
+        val resource = injector.getInstance(XtextResourceSet).getResource(URI.createURI(uri), true)
+        val elkNode = kNode.getProperty(KlighdInternalProperties.MODEL_ELEMEMT)
+        if (elkNode instanceof ElkNode) {
+            elkNode.setProperty(RectPackingOptions.ASPECT_RATIO, constraint.aspectRatio)
+            val elkGraph = EcoreUtil.getRootContainer(elkNode)
+            resource.contents.clear
+            resource.contents += elkGraph
+            resource.save(emptyMap)
+        }
+    }
+
+    /**
+     * Changes property changes defined by changedNodes to the resource
+     * @param kNode just some kNode of the correct graph
+     * @param uri uri of resource
+     * TODO use some kind of updateDocument mechanism and do not just modify the contents of the resource directly
```

Comment:
TODO.

Reply:
I will make a ticket for this instead, but note that this knid of update does not preserve the change history of the file.